pennylane/utils.py

```diff
@@ -56,32 +60,30 @@ def _unflatten(flat, model):
     See also :func:`_flatten`.
 
     Args:
-        flat (array): 1D array of items
-        model (array, Iterable, Number): model nested structure
+        flat (array[Any]): 1D array of items
+        model (Iterable, Any): model nested structure
 
     Raises:
         TypeError: if ``model`` contains an object of unsupported type
 
     Returns:
-        Union[array, list, Any], array: first elements of flat arranged into the nested
+        Union[array, list, Any], array: first elements of `flat` arranged into the nested
         structure of model, unused elements of flat
     """
-    if isinstance(model, (numbers.Number, VariableRef, str)):
-        return flat[0], flat[1:]
-
     if isinstance(model, np.ndarray):
         idx = model.size
         res = np.array(flat)[:idx].reshape(model.shape)
         return res, flat[idx:]
 
-    if isinstance(model, Iterable):
+    if isinstance(model, Iterable) and not isinstance(model, (str, bytes)):
         res = []
         for x in model:
             val, flat = _unflatten(flat, x)
             res.append(val)
         return res, flat
 
-    raise TypeError("Unsupported type in the model: {}".format(type(model)))
```

Comment:
Note that this error would arise _a lot_, in cases where torch and tensorflow variables would accidentally find themselves in a QNode without being unwrapped. Does removing this exception and replacing it with `return flat[0], flat[1:]` cause usability issues?

Reply:
I haven't found any yet. This way `_flatten` and `_unflatten` are inverse operations as one would assume, since `_flatten` does not have any unsupported types.